packages/gatsby/src/commands/develop-process.ts

```diff
@@ -135,31 +131,56 @@ module.exports = async (program: IProgram): Promise<void> => {
     initial: `initializing`,
     states: {
       initializing: {
+        on: {
+          // Ignore mutation events because we'll be running everything anyway
+          ADD_NODE_MUTATION: undefined,
+          QUERY_FILE_CHANGED: undefined,
+          WEBHOOK_RECEIVED: undefined,
+        },
         invoke: {
           src: `initialize`,
           onDone: {
             target: `initializingDataLayer`,
-            actions: `assignStoreAndWorkerPool`,
+            actions: [`assignStoreAndWorkerPool`, `spawnMutationListener`],
```

Comment:
It is part of the state machine. It's spawning it as an actor, which means it will continue to run when it changes state, and it dispatches events to the top level machine when mutations are received